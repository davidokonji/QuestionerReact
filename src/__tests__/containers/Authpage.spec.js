import React from 'react';
import { shallow } from 'enzyme';
import AuthPage from '../../containers/AuthPage.container';

let wrapper;
const props = {
  match: {
    params: []
  }
};
describe('Auth page', () => {
  beforeAll(() => {
    wrapper = shallow(<AuthPage {...props} />);
  });
  test('should render auth page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with state changes', () => {
    wrapper.setState({
      visibility: false,
      formType: 'login',
    });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('formType')).toBe('login');
  });

  it('should get onclick signup', () => {
    wrapper.find('.signup').simulate('click');
    wrapper.find('.login').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  it('should get onclick login', () => {
    wrapper.find('.signup').simulate('click');
    wrapper.find('.login').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
