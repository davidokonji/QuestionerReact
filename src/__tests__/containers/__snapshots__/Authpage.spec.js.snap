// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth page should get onclick login 1`] = `
<Fragment>
  <withRouter(Header)
    navClass="nav_header_light"
  />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={1000}
      bodyClassName={null}
      className={null}
      closeButton={false}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={true}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="wrapper_container"
    >
      <div
        className="signup"
        id="signup"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#e9eaea",
          }
        }
      >
        Sign Up
      </div>
      <div
        className="login"
        id="login"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        Sign In
      </div>
      <withRouter(Connect(LoginComponent)) />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth page should get onclick signup 1`] = `
<Fragment>
  <withRouter(Header)
    navClass="nav_header_light"
  />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={1000}
      bodyClassName={null}
      className={null}
      closeButton={false}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={true}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="wrapper_container"
    >
      <div
        className="signup"
        id="signup"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#e9eaea",
          }
        }
      >
        Sign Up
      </div>
      <div
        className="login"
        id="login"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        Sign In
      </div>
      <withRouter(Connect(LoginComponent)) />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth page should render auth page correctly 1`] = `
<Fragment>
  <withRouter(Header)
    navClass="nav_header_light"
  />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={1000}
      bodyClassName={null}
      className={null}
      closeButton={false}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={true}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="wrapper_container"
    >
      <div
        className="signup"
        id="signup"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        Sign Up
      </div>
      <div
        className="login"
        id="login"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#e9eaea",
          }
        }
      >
        Sign In
      </div>
      <withRouter(Connect(Register)) />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth page should render with state changes 1`] = `
<Fragment>
  <withRouter(Header)
    navClass="nav_header_light"
  />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={1000}
      bodyClassName={null}
      className={null}
      closeButton={false}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={true}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="wrapper_container"
    >
      <div
        className="signup"
        id="signup"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#e9eaea",
          }
        }
      >
        Sign Up
      </div>
      <div
        className="login"
        id="login"
        onClick={[Function]}
        role="presentation"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        Sign In
      </div>
      <withRouter(Connect(LoginComponent)) />
    </div>
  </div>
  <Footer />
</Fragment>
`;
